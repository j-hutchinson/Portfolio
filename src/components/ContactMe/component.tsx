import {
  StyledContainer,
  StyledDescription,
  StyledLeft,
  StyledLink,
  StyledTitle
} from './styles';

interface Props {
  theme: Theme;
}

export const ContactMe: React.FC<Props> = ({ theme }): JSX.Element => (
  <StyledContainer>
    <StyledLeft>
      <StyledTitle theme={theme}>
        <div>
          Contact Me
        </div>
      </StyledTitle>
      <StyledDescription>Want to see more of my work or get in touch. Get in contact through my socials ✌️</StyledDescription>
      <div style={{ display: 'flex', gap: "16px" }}>
        <StyledLink href="https://github.com/j-hutchinson" target="_blank" rel="noopener noreferrer">More work</StyledLink>
        <StyledLink href="mailto:jackhutchy@hotmail.com?subject=Hey, I saw your site" target="_blank" rel="noopener noreferrer">Contact</StyledLink>
      </div>
    </StyledLeft>
  </StyledContainer>
);
