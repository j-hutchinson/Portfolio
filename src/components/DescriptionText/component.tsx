import styled from 'styled-components'

const StyledDescription = styled.p`
    display: block;
    font-size: 14px;
    margin: 8px 0px 20px 0px;
    line-height: 1.47059;
    font-weight: 300;
    margin-bottom: 0;
`;

export const Description = ({ desc }) => (
    <StyledDescription>{desc}</StyledDescription>
);