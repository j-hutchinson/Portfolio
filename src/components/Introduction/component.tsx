import { Socials } from '../Socials';
import space from './spaceman.jpg';
import {
    StyledWrapper,
    StyledText,
    StyledTextContainer,
    StyledHeading,
    StyledH2,
    StyledPara,
    StyledLink,
    StyledImg
} from './Introduction.styles';

export const Introduction: React.FC = (): JSX.Element => (
    <StyledWrapper>
        <StyledText>
            <StyledTextContainer>
                <StyledHeading>Jack Hutchinson</StyledHeading>
                <StyledH2>
                    Senior Front end engineer at{' '}
                    <StyledLink href="https://checkout.com" target="_blank" rel="noopener noreferrer">
                        Checkout.com
                    </StyledLink>
                </StyledH2>
                <StyledPara>
                    Building reliable and scalable systems by day. Leveraging technology to launch fully featured products by night.
                    Design systems · A11y · Javascript
                </StyledPara>
                <Socials />
            </StyledTextContainer>
            <StyledImg
                alt="Spaceman"
                height="480px"
                width="343px"
                src={space}
                loading="lazy" />
        </StyledText>
    </StyledWrapper >
);
