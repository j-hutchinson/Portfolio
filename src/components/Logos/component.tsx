export const CFGLogo = () => (
    <svg xmlns="http://www.w3.org/2000/svg" width="150" height="204">
        <g fill="none" fillRule="evenodd">
            <path className="logo" d="M0 18.01v28.87h34.74V0H17.92A19.33 19.33 0 010 18.01M48.97 197.8h94.7V6.2h-94.7v191.6zM143.67 0H42.76v204h107.1V0h-6.2z" fill="#FFFFFE" />
            <path className="logo" fill="#FFFFFE" d="M135.7 41.28h-9.14V32.4h8.07v-5.5h-8.07v-7.14h9.14v-5.54h-15.77v32.57h15.77v-5.51M73.26 55.57v-5.54H57.23V82.6h6.63V68.53h8.33V63h-8.33v-7.42h9.4M76.33 82.6h6.25V50.03h-6.25zM124.67 82.25h6.26V55.22h4.77v-5.54h-15.77v5.54h4.74v27.03M63.22 118.57c2 0 4.1-.61 5.13-2.9.05.69.15 1.84.3 2.41h4.7v-18.6h-8.44v5.28h2.15v4.67c0 2.68-.52 3.86-2.04 3.86-1.25 0-1.96-.8-1.96-3.02v-16.7c0-2.12.53-3.17 1.96-3.17 1.74 0 1.93 1.46 1.93 3.04v1.8h6.3v-1.47c0-3.28-.75-8.74-8.25-8.74-6.21 0-8.28 4.13-8.28 9.13v15.32c0 5.32 1.43 9.1 6.5 9.1M76.33 118.08h6.25V85.52h-6.25zM110.26 85.52h-6.63v32.56h13.95v-5.48h-7.32V85.52M127.7 118.57c6.47 0 8.21-4.2 8.21-9.3 0-5.36-1.72-7.48-6.5-11.24-2.7-2.1-3.06-3.07-3.06-5.03 0-1.84.38-2.84 1.53-2.84 1.24 0 1.57.98 1.57 2.96v1.95h5.91v-1.88c0-3.73-.97-8.2-7.52-8.2-6.17 0-7.7 4.3-7.7 8.95 0 4.8 1.88 6.63 5.55 9.74 3.32 2.78 3.9 3.67 3.9 6.75 0 1.53-.34 3-1.78 3-1.43 0-1.96-.99-1.96-3.13v-2.25h-6.14V110c0 4.82 1.78 8.57 7.99 8.57M65.36 47.27c6.21 0 8.64-4.13 8.64-9.12v-2.06h-6.7v2.64c0 2.12-.53 3.17-1.96 3.17-1.74 0-1.93-1.46-1.93-3.04l-.01-1.75V22.3c0-2.12.57-3.2 2-3.2 1.74 0 1.93 1.46 1.93 3.03v2.43H74v-2.1c0-3.28-1.12-8.74-8.62-8.74-6.21 0-8.63 4.13-8.63 9.13v15.67c0 3.29 1.1 8.74 8.6 8.74M94.87 95.3c0 3.03-.85 3.69-2.94 3.69h-.42v-8.54h.62c2.3 0 2.74 1.09 2.74 3.27v1.58zm-2.22-9.78H85.6v32.56h5.9v-14.15h.38c2.39 0 2.8.62 2.8 4.28v9.87h5.86v-9.6c0-4.55-.95-6.04-2.97-7.02 1.4-.66 3.25-2.04 3.25-6.68v-1.17c0-5.7-2.7-8.1-8.16-8.1zM84.04 22.23c0-2.16.8-3.13 2.27-3.13 1.72 0 2.34 1.04 2.34 3.08v16.66c0 2.14-.73 3.1-2.33 3.1-1.53 0-2.28-1.06-2.28-3.07V22.23zm2.02 25.04c6.23 0 9.56-2.77 9.56-9.23v-15.1c0-6.51-3.72-9.2-9.25-9.2-5.49 0-9.27 2.74-9.27 9.02V38.1c0 6.33 3.04 9.17 8.96 9.17zM105.19 41.74V19.3h1.44c2.67 0 3.74 1.3 3.74 3.71V37.6c0 2.54-.92 4.14-3.69 4.14h-1.49zm11.89-5.48V24.02c0-6.74-2.6-9.8-10.17-9.8H98.6v32.57h8.8c8.34 0 9.68-4.92 9.68-10.53zM94.87 59.82c0 3.03-.85 3.69-2.94 3.69h-.42v-8.54h.62c2.3 0 2.74 1.09 2.74 3.27v1.58zm-2.22-9.79H85.6V82.6h5.9V68.44h.38c2.39 0 2.8.63 2.8 4.3v9.86h5.86V73c0-4.55-.95-6.04-2.97-7.02 1.4-.66 3.25-2.04 3.25-6.68v-1.17c0-5.7-2.7-8.1-8.16-8.1zM117.97 59.67V57.8c0-3.73-.93-8.2-7.17-8.2-5.89 0-7.35 4.29-7.35 8.94 0 4.8 1.8 6.64 5.3 9.75 3.16 2.78 3.71 3.67 3.71 6.74 0 1.54-.32 3.01-1.7 3.01-1.36 0-1.86-.99-1.86-3.13v-2.26h-5.85v1.95c0 4.83 1.69 8.58 7.61 8.58 6.17 0 7.83-4.2 7.83-9.3 0-5.37-1.64-7.48-6.2-11.24-2.57-2.1-2.91-3.07-2.91-5.03 0-1.85.35-2.84 1.46-2.84 1.18 0 1.49.98 1.49 2.95v1.95h5.64" />
        </g>
    </svg>
);

export const MendeleyLogo = () => (
    <svg xmlns="http://www.w3.org/2000/svg" width="135" height="24">
        <g fill="none" fillRule="evenodd">
            <path className="logo" fill="#323232" d="M31.02 0c.72 0 1.41.11 2.06.32l.25.09.21.08a6.14 6.14 0 013.9 5.94 5.9 5.9 0 01-.4 2.4c-.85 4.33-.4 6.61 2.57 6.68a4.09 4.09 0 013.33 1.16s1.04 1.02 1.06 2.85c.01 1.8-.69 2.73-.96 3.02l-.06.06-.04.04-.18.17a4.32 4.32 0 01-2.44 1.16 4.1 4.1 0 01-4.6-3.7c-.07-.8.08-1.56.4-2.24 3.25-7-12.78-7.16-10.08-.55l.02.03a4.13 4.13 0 01-1.54 5.84c-.78.44-1.65.63-2.5.6a4.67 4.67 0 01-2.5-.6 4.13 4.13 0 01-1.54-5.84l.02-.03c2.71-6.61-13.32-6.45-10.07.55.31.68.46 1.44.39 2.23a4.1 4.1 0 01-4.6 3.7 4.41 4.41 0 01-2.93-1.7c-1-1.26-1.1-4.15 0-5.23s2.19-1.67 3.65-1.52C7.4 15.44 7.85 13.15 7 8.84a5.9 5.9 0 01-.41-2.41A6.14 6.14 0 0110.5.49l.22-.08.25-.09a6.34 6.34 0 017.34 2.62c1.5 1.62 2.25 2.6 3.71 2.6 1.54 0 2.22-.98 3.72-2.6A6.29 6.29 0 0131.02 0zM22.5 7C20 7 18 8.79 18 11c0 1.24.63 2.34 1.62 3.08.43.36.97.63 1.63.76.39.1.8.16 1.23.16h.04c.43 0 .84-.06 1.23-.16a3.6 3.6 0 001.63-.76A3.83 3.83 0 0027 11c0-2.21-2-4-4.48-4h-.04z" />
            <g transform="translate(52 3)">
                <path className="logo" fill="#323232" d="M11.4 1.08L9.39 5.71a110.8 110.8 0 00-2.27 5.48c-.37-1.04-1.09-2.95-2.13-5.55L3.13 1.08H.98L0 14.51h1.64l.78-11.07 4.54 11.22L11.99 3.4l.68 11.1h1.73L13.43 1.1h-2.02zm13.77 8.18zm0 0c0-3.49-.96-5.09-4.05-5.09-2.6 0-4.5 1.87-4.5 5.33 0 3.32 2 5.27 4.54 5.27A6.8 6.8 0 0025 13.8l-.14-1.55a5.46 5.46 0 01-3.16 1.04c-2.11 0-3.28-1.31-3.28-3.7v-.35h6.76zm-4.16-3.64c1.52-.02 2.54.36 2.6 2.3H18.6c.17-1.26 1.18-2.29 2.43-2.3zM57.8 9.26zm0 0c0-3.49-.96-5.09-4.05-5.09-2.6 0-4.5 1.87-4.5 5.33 0 3.32 2 5.27 4.54 5.27a6.8 6.8 0 003.83-.96l-.14-1.55a5.46 5.46 0 01-3.16 1.04c-2.11 0-3.28-1.31-3.28-3.7v-.35h6.76zm-4.16-3.64c1.52-.02 2.54.36 2.6 2.3h-5.02c.17-1.26 1.18-2.29 2.42-2.3zM72.7 9.26zm0 0c0-3.49-.96-5.09-4.05-5.09-2.6 0-4.49 1.87-4.49 5.33 0 3.32 2 5.27 4.54 5.27a6.8 6.8 0 003.82-.96l-.13-1.55a5.46 5.46 0 01-3.16 1.04c-2.12 0-3.3-1.31-3.3-3.7v-.35h6.77zm-4.15-3.64c1.51-.02 2.53.36 2.6 2.3h-5.03c.17-1.26 1.18-2.29 2.43-2.3zM32.42 4.17c-1.51 0-2.44.76-3.27 1.76l-.27-1.49h-1.57l.08.8c.07.51.1 1.2.1 2v7.26h1.7V7.84c.47-.65 1.5-2.13 3.08-2.13 1.53 0 1.67.78 1.7 2.3.03.8.03 2.14.03 2.14.02 2.02-.07 4.36-.07 4.36h1.71s.05-3.03.05-4.69c0-1 .02-2-.07-3-.18-2.07-1.4-2.65-3.2-2.65m14.63 9.54c-.07-.52-.1-1.18-.1-1.99V0h-1.7v4.62a4.53 4.53 0 00-2.2-.44c-3.14 0-5.12 1.96-5.12 5.36 0 3.13 1.9 5.24 4.3 5.24 1.33 0 2.13-.42 3.13-1.47.04.43.13.96.2 1.2h1.57l-.08-.8zm-1.8-2.16c-.58.87-1.58 1.76-2.76 1.76-1.11 0-2.69-1.25-2.69-3.93 0-1.94 1.27-3.74 3.1-3.74 1.62 0 2.35.45 2.35 1.87v4.04zm12.56-2.3v.02-.01z" />
                <mask className="logo" id="b" fill="#fff">
                    <path id="a" d="M0 18.95h82.86V0H0z" />
                </mask>
                <path className="logo" fill="#323232" d="M60.12 14.5h1.7V0h-1.7zm12.56-5.24zm8.5-5.09l-2.74 8-3.07-8h-1.83l4.05 10.45-.54 1.26c-.57 1.4-1.35 3.07-1.35 3.07h1.83s.73-1.84 1.48-3.97l3.85-10.8h-1.69" />
            </g>
        </g>
    </svg>
);

export const CheckoutLogo = () => (
    <svg xmlns="http://www.w3.org/2000/svg" width="185" height="24" fill="none">
        <path className="logo" d="M137.26 18.92c1 0 1.83-.83 1.83-1.86 0-1.03-.82-1.86-1.83-1.86-1 0-1.82.83-1.82 1.86 0 1.03.81 1.86 1.82 1.86zM32.27 18.43a5.34 5.34 0 01-2.07-1.47 7.28 7.28 0 01-1.82-4.93c0-.98.14-1.9.43-2.74a5.9 5.9 0 011.3-2.2 5.99 5.99 0 012.01-1.47 6.78 6.78 0 012.7-.53c.86 0 1.58.1 2.25.34.67.24 1.25.58 1.78.98.53.39.91.92 1.2 1.51.33.59.53 1.27.62 1.96h-3.36a2.49 2.49 0 00-.82-1.47c-.43-.4-.96-.59-1.58-.59-1.06 0-1.83.4-2.35 1.13a5.14 5.14 0 00-.77 3.03c0 1.27.24 2.25.77 3.03.48.78 1.3 1.12 2.35 1.12.72 0 1.3-.2 1.73-.53.43-.4.72-.93.86-1.72h3.36a5.36 5.36 0 01-3.5 4.65 7.2 7.2 0 01-2.35.39c-1.01.05-1.93-.15-2.74-.49zM46.24 7.09c.48-.68 1.01-1.22 1.59-1.56.57-.35 1.3-.54 2.2-.54a4.36 4.36 0 013.23 1.42 5.31 5.31 0 011.25 3.57v8.6h-3.46v-8.12c0-.73-.2-1.32-.58-1.76a2.15 2.15 0 00-1.68-.68c-.77 0-1.4.3-1.87.83a3.2 3.2 0 00-.77 2.2v7.53h-3.46V.15h3.46v6.94h.1zM60.17 18.43a5.9 5.9 0 01-3.4-3.72 8.6 8.6 0 01.04-5.47c.34-.83.77-1.57 1.3-2.2a5.98 5.98 0 014.6-2 5.97 5.97 0 014.76 2.2c.53.68.96 1.56 1.25 2.49.29.98.43 2.05.43 3.22h-9.41a4.53 4.53 0 001 2.45c.54.58 1.3.88 2.31.88.68 0 1.2-.15 1.64-.44.38-.3.72-.69.86-1.18h3.41c-.1.6-.33 1.13-.62 1.67a5.15 5.15 0 01-2.98 2.3c-.67.24-1.44.34-2.26.34-1.1 0-2.06-.2-2.93-.54zm.53-9.97a4.28 4.28 0 00-.96 2.05h5.76a3.28 3.28 0 00-.81-2.05 2.48 2.48 0 00-1.92-.79c-.92.05-1.59.3-2.07.79zM74.3 18.43a5.34 5.34 0 01-2.07-1.47 7.27 7.27 0 01-1.83-4.93c0-.98.15-1.9.43-2.74a5.9 5.9 0 011.3-2.2 5.99 5.99 0 012.02-1.47 6.78 6.78 0 012.69-.53c.86 0 1.58.1 2.26.34.67.24 1.24.58 1.77.98.53.39.92.92 1.2 1.51.34.59.53 1.27.63 1.96h-3.36a2.49 2.49 0 00-.82-1.47c-.43-.4-.96-.59-1.59-.59-1.05 0-1.82.4-2.35 1.13a5.14 5.14 0 00-.77 3.03c0 1.27.24 2.25.77 3.03.48.78 1.3 1.12 2.35 1.12.72 0 1.3-.2 1.73-.53.44-.4.72-.93.87-1.72h3.31a5.36 5.36 0 01-3.5 4.65c-.72.24-1.5.39-2.36.39a6.14 6.14 0 01-2.69-.49zM88.17 14.13v4.45h-3.4V.15h3.4v10.21l4.61-4.98h4.14l-4.8 5.03 5.37 8.17h-4.18l-3.65-6.02-1.49 1.57zM101.28 18.43a6.55 6.55 0 01-2.16-1.47 7.4 7.4 0 01-1.39-2.2 7.28 7.28 0 01-.48-2.73 6.83 6.83 0 011.87-4.94 5.7 5.7 0 012.16-1.47 7.03 7.03 0 012.74-.53c1.01 0 1.92.2 2.74.53.82.35 1.54.83 2.16 1.47a7.4 7.4 0 011.4 2.2c.33.83.48 1.76.48 2.74a6.83 6.83 0 01-1.87 4.94 5.7 5.7 0 01-2.17 1.46c-.82.34-1.73.54-2.74.54-1 0-1.92-.2-2.73-.54zm5.2-3.32c.57-.79.86-1.81.86-3.13 0-1.32-.3-2.35-.87-3.13a2.9 2.9 0 00-2.45-1.22c-1.05 0-1.87.39-2.45 1.22a5.22 5.22 0 00-.86 3.13c0 1.27.29 2.34.86 3.13.58.78 1.4 1.17 2.45 1.17 1.06 0 1.88-.4 2.45-1.17zM120.88 17.01h-.1a5.81 5.81 0 01-1.48 1.42c-.58.34-1.3.49-2.21.49a4.49 4.49 0 01-3.37-1.27c-.76-.88-1.15-2-1.15-3.47v-8.8h3.41v8.21c0 .78.2 1.37.53 1.76.34.4.91.64 1.63.64.82 0 1.44-.3 1.92-.84.48-.53.72-1.27.72-2.15V5.38h3.46v13.2h-3.31V17h-.05zM127.56 5.38V1.22h3.36v4.16h2.2v2.3h-2.2v6.98c0 .5.1.84.34 1.03.24.2.52.3.96.3h.67c.1 0 .24 0 .34-.05v2.63c-.2 0-.34.05-.58.05-.2 0-.43.05-.67.05h-.92c-.43 0-.81-.05-1.25-.14-.43-.1-.76-.25-1.1-.5a2.68 2.68 0 01-.82-.97c-.19-.4-.33-.93-.33-1.56V7.67h-1.78v-2.3h1.78zM144.08 18.38a5.47 5.47 0 01-1.87-1.47 6 6 0 01-1.16-2.2 10 10 0 010-5.57 7.3 7.3 0 011.16-2.2 5.47 5.47 0 014.41-2c1.44 0 2.65.39 3.56 1.12.91.73 1.49 1.81 1.73 3.18h-1.54c-.2-.98-.58-1.76-1.25-2.25a3.86 3.86 0 00-2.45-.73c-.77 0-1.39.14-1.97.44-.52.29-1 .68-1.34 1.22a5.1 5.1 0 00-.77 1.8c-.2.7-.24 1.43-.24 2.2 0 .79.1 1.52.29 2.2.19.7.43 1.28.82 1.82.33.53.81.92 1.34 1.22a4 4 0 001.92.44 3.8 3.8 0 002.69-.98 4.2 4.2 0 001.2-2.6h1.49a5.5 5.5 0 01-1.63 3.63 5.17 5.17 0 01-3.75 1.32 6.83 6.83 0 01-2.64-.59zM156.9 18.38a5.2 5.2 0 01-1.92-1.47 6.92 6.92 0 01-1.2-2.2 8.47 8.47 0 01-.43-2.78c0-1.03.14-1.9.43-2.79.29-.83.67-1.61 1.2-2.2a5.2 5.2 0 011.92-1.46 6.09 6.09 0 012.55-.54 5.63 5.63 0 014.47 2c.52.64.9 1.37 1.2 2.2.28.83.38 1.76.38 2.79 0 1.02-.14 1.9-.38 2.78a6.44 6.44 0 01-1.2 2.2 5.21 5.21 0 01-1.93 1.47 6.25 6.25 0 01-5.1 0zm4.47-1.22c.58-.3 1-.69 1.4-1.22.38-.54.66-1.13.8-1.81a8.11 8.11 0 000-4.4 6.85 6.85 0 00-.8-1.81 3.74 3.74 0 00-1.4-1.22c-.58-.3-1.2-.44-1.92-.44a4 4 0 00-1.92.44c-.53.29-1.01.68-1.4 1.22a5.7 5.7 0 00-.86 1.8c-.2.7-.29 1.43-.29 2.2 0 .79.1 1.52.29 2.2.2.7.48 1.28.86 1.82a4 4 0 003.31 1.66 4 4 0 001.93-.44zM169.3 5.23v2h.04c.44-.68.96-1.22 1.59-1.66a4.1 4.1 0 012.4-.63c.82 0 1.54.2 2.16.59.63.39 1.06 1.02 1.34 1.8h.05a4.59 4.59 0 015.72-2.15c.48.2.91.44 1.25.79.33.34.62.78.81 1.32.2.53.3 1.12.3 1.76v9.53h-1.55V9.24c0-.98-.23-1.71-.76-2.2a2.78 2.78 0 00-1.97-.68c-.48 0-.96.1-1.35.24-.43.2-.81.44-1.1.73-.29.3-.58.74-.77 1.18-.2.49-.29.98-.29 1.56v8.5h-1.58V9.25c0-.98-.24-1.71-.77-2.2a2.48 2.48 0 00-1.88-.68c-.47 0-.9.1-1.34.29a3.6 3.6 0 00-2.02 2c-.19.44-.29.98-.29 1.52v8.4h-1.58V5.24h1.58zM15.4 12.02l6.16-10.94a.68.68 0 000-.74.82.82 0 00-.63-.34H8.25a.82.82 0 00-.62.34L1.1 11.64c-.15.24-.15.53 0 .73l6.53 11.29a.7.7 0 00.62.34h12.68c.24 0 .48-.15.63-.34a.68.68 0 000-.74l-6.15-10.9zm-.8-1.46l-5.1-9.1h10.23l-5.14 9.1zM8.2 2.2l5.53 9.82-5.53 9.83-5.66-9.83L8.2 2.2zm1.3 20.33l5.1-9.09 5.08 9.1H9.5z" fill="white" />
    </svg>
);

export const TrayLogo = () => (
    <svg xmlns="http://www.w3.org/2000/svg" fill="none" viewBox="0 0 610 240">
        <path fill="white" className="logo" d="M234.44 118.223v-17.07h-24.52v-19.69h-19.95v19.69h-20.72v17.07h20.72v25.779c0 12.179 2.34 19.459 7.38 24.499 6.85 6.86 16.09 8.54 23.89 8.54 9.34 0 12.35-1.53 12.35-1.53v-16.84a75.28 75.28 0 01-11.6.76c-2.37 0-6.53-.53-9.22-3.22-2.34-2.34-2.89-6.219-2.89-9.479v-28.509h24.56zm55.31-18.61c-8.36 0-15.77 2.93-20.33 7.49a20.023 20.023 0 00-1.41 1.56v-7.51h-19.17v74.388h19.91v-46.539a22.758 22.758 0 0119.47-10.999c2.74.019 5.474.276 8.17.77v-18.54a43.525 43.525 0 00-6.64-.62zm181.86 1.54h-21l-20.24 44.339-21.63-44.339H386.8l33.29 66.868v.01L405.02 200h20.85l45.74-98.847zm-112.3 0v5.9a36.171 36.171 0 00-21.76-7.44c-20.1 0-36.57 16.69-36.57 38.809s17 38.649 37.47 38.649c12.75 0 20.12-6.73 20.82-7.39v5.86h19.14v-74.388h-19.1zm-18.76 57.518c-10.71 0-19.61-8.149-19.61-20.249 0-11.57 8.24-20.399 19.15-20.399a23.761 23.761 0 0118.41 8.899v22.87a22.297 22.297 0 01-17.95 8.879zm130.12-7.749c2.574.057 5.074.872 7.186 2.344a13.074 13.074 0 015.316 13.46 13.08 13.08 0 01-25.872-2.724 13.104 13.104 0 018.279-12.164 13.095 13.095 0 015.091-.916zm36.69-87.467a12.742 12.742 0 0112.203 15.381 12.741 12.741 0 01-25.203-2.641 12.785 12.785 0 018.051-11.835 12.779 12.779 0 014.949-.905zm9.96 37.698h-19.91v74.388h19.91v-74.388zm52.94-1.54c-21.86 0-39.58 16.87-39.58 38.729s17.72 38.739 39.58 38.739 39.59-16.87 39.59-38.739c0-21.869-17.73-38.729-39.59-38.729zm0 59.058c-10.85 0-19.64-8.209-19.64-20.329 0-11.92 8.79-20.319 19.64-20.319 10.85 0 19.66 8.399 19.66 20.319 0 12.12-8.81 20.329-19.66 20.329zM99.55 40.005a39.82 39.82 0 00-39.82 39.819v7.47h19.91v-7.47a19.91 19.91 0 1119.91 19.92H39.82a39.819 39.819 0 100 79.637h7.46v-19.91h-7.46a19.909 19.909 0 110-39.818h19.91v59.728h19.91v-59.728h19.91a39.84 39.84 0 0037.369-24.437 39.833 39.833 0 00-8.798-43.773A39.825 39.825 0 0099.55 40.005z"></path>
    </svg>
);