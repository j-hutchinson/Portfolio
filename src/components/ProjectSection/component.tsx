import {
    StyledTitle,
    StyledContainer,
    StyledDescription,
    StyledImg,
    StyledLink,
    StyledLeft,
    StyledTools
} from './styles';
import colorScheme from './colorScheme.png';

interface Props {
    desc: string;
    name: string;
    link: string;
}

const ProjectSection = ({ name, desc, link }: Props) => (
    <StyledContainer>
        <StyledLeft>
            <div style={{ display: 'flex', justifyContent: 'space-between', flexDirection: 'column' }}>
                <StyledTitle>
                    <div>
                        {name}
                    </div>
                </StyledTitle>
                <StyledTools>React Â· Typescript </StyledTools>
                <StyledDescription>{desc}</StyledDescription>
            </div>
            <StyledLink href={link} target="_blank" rel="noopener noreferrer">Github</StyledLink>
        </StyledLeft>
        <StyledImg src={colorScheme} />
    </StyledContainer>
)

export default ProjectSection;