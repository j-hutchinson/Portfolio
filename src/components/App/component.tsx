import styled, { ThemeProvider } from 'styled-components';
import { GlobalStyles } from "../../css/GlobalStyle";
import { darkTheme, lightTheme } from "../../css/tokens";
import { Checkout, ColorScheme, Mendeley, CodeFirstGirls } from '../constants';
import { ContactMe } from '../ContactMe';
import { Introduction } from '../Introduction';
import { JobSection } from '../JobSection';
import { ProjectSection } from '../ProjectSection';
import { Toggle, useDarkMode } from "../ThemeToggle";

const StyledBody = styled.main`
  padding-bottom: 64px;
`;

const App = (): JSX.Element => {
  const [theme, themeToggler] = useDarkMode();
  const themeMode = theme === 'light' ? lightTheme : darkTheme;

  return (
    <ThemeProvider theme={themeMode}>
      <>
        <GlobalStyles />
        <Toggle toggleTheme={themeToggler} />
        <Introduction />
        <StyledBody >
          <JobSection {...Checkout} theme={themeMode} />
          <JobSection {...Mendeley} theme={themeMode} />
          <JobSection {...CodeFirstGirls} theme={themeMode} />
          <ProjectSection {...ColorScheme} theme={themeMode} />
          <ContactMe theme={themeMode} />
        </StyledBody>
      </>
    </ThemeProvider >
  )
};

export default App;
