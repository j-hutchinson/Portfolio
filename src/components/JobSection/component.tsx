import { Jobs } from '../constants';
import { CFGLogo, CheckoutLogo, MendeleyLogo } from '../Logos/component';
import { Description } from '../DescriptionText';
import {
    StyledCompanyName,
    StyledContainer,
    StyledLink,
    StyledLogo,
    StyledLeft,
    StyledTitle
} from './styles';

interface Props {
    desc: string;
    endDate?: string;
    startDate?: string;
    name: string;
    title: string;
    link: string;
    theme: Theme;
}

export const JobSection: React.FC<Props> = ({ name, title, desc, link, theme }): JSX.Element => (
    <StyledContainer>
        <StyledLeft>
            <StyledCompanyName>
                <StyledLink href={link} target="_blank" rel="noopener noreferrer" className="company-link">
                    {name}
                </StyledLink>
            </StyledCompanyName>
            <StyledTitle theme={theme}>
                <span>
                    {title}
                </span>
            </StyledTitle>
            <Description desc={desc} />
        </StyledLeft>
        <StyledLogo>
            {name === Jobs.Mendeley && <MendeleyLogo />}
            {name === Jobs.CFG && <CFGLogo />}
            {name === Jobs.Checkout && <CheckoutLogo />}
        </StyledLogo>
    </StyledContainer>
);
