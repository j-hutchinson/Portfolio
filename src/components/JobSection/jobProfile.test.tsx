import { shallow } from 'enzyme';
import { darkTheme } from '../../css/tokens';
import { JobSection } from './component';

describe('JobSection component', () => {
    const profile = {
        desc: "desc",
        endDate: "endDate",
        startDate: "startDate",
        name: "name",
        title: "title",
        link: 'somewhere_over_the_rainbow',
        logo: <div>logo</div>
    }
    test('component matches snapshot with full profile', () => {
        expect.assertions(1);

        const wrapper = shallow(<JobSection {...profile} theme={darkTheme} />);

        expect(wrapper).toMatchSnapshot();
    });
    test('component matches snapshot with no endDate', () => {
        expect.assertions(1);

        const wrapper = shallow(<JobSection {...profile} endDate={null} theme={darkTheme} />);

        expect(wrapper).toMatchSnapshot();
    });
});