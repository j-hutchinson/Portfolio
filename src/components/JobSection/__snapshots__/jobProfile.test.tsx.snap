// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JobSection component component matches snapshot with full profile 1`] = `
<styled.section>
  <styled.div>
    <styled.h3>
      <styled.a
        className="company-link"
        href="somewhere_over_the_rainbow"
        rel="noopener noreferrer"
        target="_blank"
      >
        name
      </styled.a>
    </styled.h3>
    <styled.h4
      theme={
        Object {
          "background": "linear-gradient(to bottom right, #383838, #1E1E1E)",
          "border": "#EDEFE7",
          "fontColor": "white",
          "highlight": "#5198db",
          "imgShadow": "rgb(0 0 0 / 80%)",
          "primaryBackground": "#222",
          "secondarFontColor": "#AFB0B3",
          "secondaryBackground": "radial-gradient(118.99% 670.46% at -7.06% -9.5%,#191919 0%,#0a0a0a 100%)",
          "sectionBackground": "radial-gradient(118.99% 670.46% at -7.06% -9.5%,#191919 0%,#0a0a0a 100%)",
          "svgFill": "white",
        }
      }
    >
      <span>
        title
      </span>
    </styled.h4>
    <Description
      desc="desc"
    />
  </styled.div>
  <styled.div />
</styled.section>
`;

exports[`JobSection component component matches snapshot with no endDate 1`] = `
<styled.section>
  <styled.div>
    <styled.h3>
      <styled.a
        className="company-link"
        href="somewhere_over_the_rainbow"
        rel="noopener noreferrer"
        target="_blank"
      >
        name
      </styled.a>
    </styled.h3>
    <styled.h4
      theme={
        Object {
          "background": "linear-gradient(to bottom right, #383838, #1E1E1E)",
          "border": "#EDEFE7",
          "fontColor": "white",
          "highlight": "#5198db",
          "imgShadow": "rgb(0 0 0 / 80%)",
          "primaryBackground": "#222",
          "secondarFontColor": "#AFB0B3",
          "secondaryBackground": "radial-gradient(118.99% 670.46% at -7.06% -9.5%,#191919 0%,#0a0a0a 100%)",
          "sectionBackground": "radial-gradient(118.99% 670.46% at -7.06% -9.5%,#191919 0%,#0a0a0a 100%)",
          "svgFill": "white",
        }
      }
    >
      <span>
        title
      </span>
    </styled.h4>
    <Description
      desc="desc"
    />
  </styled.div>
  <styled.div />
</styled.section>
`;
